plugins/module_utils/network/sonic/config/users/users.py

```diff
@@ -201,6 +199,36 @@ def _state_deleted(self, want, have, diff):
 
         return commands, requests
 
+    def _state_overridden(self, want, have, diff):
+        """ The command generator when state is overridden
+        :param want: the desired configuration as a dictionary
+        :param have: the current configuration as a dictionary
+        :param diff: the difference between want and have
+        :rtype: A list
+        :returns: the commands necessary to migrate the current configuration
+                  to the desired configuration
+        """
+        commands = []
+        requests = []
+
+        if diff:
+            # Delete all users except admin
+            commands = have
+            requests = self.get_delete_users_requests(commands, have)
+            send_requests(self._module, requests)
+            have = []
+
+            # Merge want configuration
+            commands = want
+            requests = self.get_modify_users_requests(commands, have)
```

Comment:
Override of the configuration is performed only when there is new user/modified configuration of existing user configuration in 'want'. If 'want' is a subset of 'have', then the non-specified users in 'have' are not deleted.
Please handle the above scenario.

Reply:
Addressed and added a test case for this scenario.